src/nbl/video/COpenGL_Swapchain.cpp

```diff
@@ -0,0 +1,414 @@
+
+#include "nbl/video/COpenGL_Swapchain.h"
+
+#include "nbl/video/surface/CSurfaceGL.h"
+#include "nbl/video/COpenGLSync.h"
+#include "nbl/video/IOpenGL_FunctionTable.h"
+#include "nbl/video/COpenGLFence.h"
+#include "nbl/video/COpenGLSemaphore.h"
+#include "nbl/video/COpenGLImage.h"
+#include "nbl/video/COpenGL_Queue.h"
+
+#include "nbl/system/IThreadHandler.h"
+
+namespace nbl::video
+{
+
+static inline constexpr uint32_t OpenGLFunctionTableSize = 6672u;
+using SThreadHandlerInternalState = std::array<uint8_t, OpenGLFunctionTableSize>;
+
+static_assert(OpenGLFunctionTableSize >= sizeof(COpenGLFunctionTable));
+static_assert(OpenGLFunctionTableSize >= sizeof(COpenGLESFunctionTable));
```

Comment:
you know that `std::max` is constexpr now?

you could just set it to a max of both `sizeof`